### ISSUE TITLE
Interactive highlight can disappear

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/mean-share-and-balance#125]
**Test device**
iPad 9th generation

**Operating System**
iPadOS 16

**Browser**
Safari

**Problem description**
For https://github.com/phetsims/qa/issues/848, when moving the pencil or water level arrow up and down repeatedly, the interactive highlight sometimes disappears. I was able to produce this on 2 different ipads (both with the latest iOS).

**Steps to reproduce**
It isn't that easy to reproduce, but this is what I was doing: 
On an iPad:
1. Turn on interactive highlights
2. Close valve
3. Check Predict Mean
4. Lightly grab the pencil and move it up and down, lifting your finger every once in a while. 

**Visuals**

https://user-images.githubusercontent.com/87318828/200128905-102c4c45-6150-4a05-8611-fa2cdd1fa8a7.mov

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Mean: Share and Balance‬
URL: https://phet-dev.colorado.edu/html/mean-share-and-balance/1.0.0-rc.1/phet/mean-share-and-balance_all_phet.html
Version: 1.0.0-rc.1 2022-11-01 17:03:54 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36
Language: en-US
Window: 2079x970
Pixel Ratio: 1.3333333730697632/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}
</details>

**Comment by jessegreenberg** [comment]
This looks like a general problem with `touchSnag`. An easy way to reproduce this problem is to put your finger down to the right of the pencil, then drag your finger over it. Then drag your finger back to the right. The pencil will still be dragged but the highlight will not stay around it.

**Comment by jessegreenberg** [comment]
I believe this is fixed from the above commit, @Nancy-Salpepi would you mind confirming?

**Comment by Nancy-Salpepi** [comment]
I was no longer able to reproduce this issue @jessegreenberg.

**Comment by marlitas** [comment]
Awesome. Thanks all! Marking for cherry-pick and keeping open for the next RC.

**Comment by jessegreenberg** [comment]
Sorry, I believe there is a problem from this change. @KatieWoe reported some sims are failing from multitouch fuzzing. There may be another commit to cherry-pick.

**Comment by jessegreenberg** [comment]
Should be fixed with the above commit. Id like to make sure that CT is clear before reassigning back for cherry-pick.

**Comment by jessegreenberg** [comment]
I am sorry to see this is still a problem on CT. This needs a bit more effort I am afraid. I am going to revert the changes above and create a new issue in scenery to investigate. In the meantime I would recommend publishing the prototype with this issue and not letting it get blocked by this. @marlitas feel free to confirm that is OK though with someone on design.

**Comment by samreid** [comment]
@jessegreenberg recommendation in https://github.com/phetsims/mean-share-and-balance/issues/125#issuecomment-1309616678 seems reasonable to me.  Also, in https://github.com/phetsims/scenery/issues/1495 @jessegreenberg said:

> I tried a solution in that issue but it didn't work (ended up reverting). 

But I do not see the revert commit, or any commit referenced after https://github.com/phetsims/mean-share-and-balance/issues/125#issuecomment-1309616678

**Comment by jessegreenberg** [comment]
> But I do not see the revert commit, or any commit referenced after

Thanks! Sorry, failed to push.

**Comment by amanda-phet** [comment]
Prototypes can definitely have this kind of behavior present, so I am fine with moving ahead with the prototype.

**Comment by marlitas** [comment]
Assigning to @jbphet and @marlitas to look at in upcoming development.

**Comment by marlitas** [comment]
I looked into this a bit further and I believe it would be beneficial to meet with @jessegreenberg to discuss prior to attempting a solution. The code @jessegreenberg reverted would be my initial approach as well, and I don't currently see a way to move forward without retracing his steps, therefore his insight would be useful in not doubling up that time.

**Comment by marlitas** [comment]
Messaged @jessegreenberg on slack and he said:

> I think those [reverted] commits are on the right track but were causing CT problems and I am not sure why yet. I think the next step is to apply them and debug from the errors. Just reviewing those commits I don't see a 'removeInputListener' anywhere for the attached Pointer listener, maybe thats it.

So I'm going to go ahead and bring those reverted commits back and track/debug errors from there to start.

**Comment by zepumph** [comment]
This is still failing many sims on CT. Is there anything I can do to help with this?

>Uncaught Error: Assertion failed: It should be impossible to already have a Pointer before locking from touchSnag

**Comment by marlitas** [comment]
@jbphet and I tested this after @jessegreenberg's fix for https://github.com/phetsims/scenery/issues/1495, and it's looking great! Closing.
